code/controllers/subsystem/spatial_gridmap.dm

```diff
@@ -0,0 +1,256 @@
+ */
+
+SUBSYSTEM_DEF(spatial_grid)
+	can_fire = FALSE
+	init_order = INIT_ORDER_SPATIAL_GRID
+	name = "Spatial Grid"
+
+	///list of the spatial_grid_cell datums per z level, arranged in the order of y index then x index
+	var/list/grids_by_z_level = list()
+
+/datum/controller/subsystem/spatial_grid/Initialize(start_timeofday)
+	. = ..()
+	var/cells_per_side = world.maxx / SPATIAL_GRID_CELLSIZE //assume world.maxx == world.maxy
+	for(var/datum/space_level/z_level as anything in SSmapping.z_list)
+		var/list/new_cell_grid = list()
+
+		grids_by_z_level += list(new_cell_grid)
+
+		for(var/y in 1 to cells_per_side)
+			new_cell_grid += list(list())
+			for(var/x in 1 to cells_per_side)
+				var/datum/spatial_grid_cell/cell = new(x, y, z_level.z_value)
+				new_cell_grid[y] += cell
+
+/**
+ * searches through the grid cells intersecting range radius around center and returns the added contents that are also in LOS
+ * much faster than iterating through view() to find all of what you want for things that arent that common
+ *
+ * * center - the atom that is the center of the searched circle
+ * * type - the grid contents channel you are looking for, see __DEFINES/spatial_grid.dm
+ * * range - the radius of our search circle. the code assumes this is > 1
+ * * ignore_visibility - if TRUE, line of sight is ignored, the contents of the grid are only filtered for distance
+ * * include_center - if FALSE, subtracts center from the output before filtering, used to speedup searches where you dont care about center being in the output
+ */
+/datum/controller/subsystem/spatial_grid/proc/find_grid_contents_in_view(atom/center, type, range, ignore_visibility = FALSE, include_center = TRUE)//should probably just be a global proc but w/e
+	var/turf/center_turf = get_turf(center)
+
+	var/center_x = center_turf.x
+	var/center_y = center_turf.y
+
+	var/list/contents_to_return = list()
+
+	var/static/grid_cells_per_axis = world.maxx / SPATIAL_GRID_CELLSIZE//im going to assume this doesnt change at runtime
+
+	//the minimum x and y cell indexes to test
+	var/min_x = max(CEILING((center_x - range) * (1 / SPATIAL_GRID_CELLSIZE), 1), 1)
+	var/min_y = max(CEILING((center_y - range) * (1 / SPATIAL_GRID_CELLSIZE), 1), 1)//calculating these indices only takes around 2 microseconds
+
+	//the maximum x and y cell indexes to test
+	var/max_x = min(CEILING((center_x + range) * (1 / SPATIAL_GRID_CELLSIZE), 1), grid_cells_per_axis)
+	var/max_y = min(CEILING((center_y + range) * (1 / SPATIAL_GRID_CELLSIZE), 1), grid_cells_per_axis)
+
+	var/list/grid_level = grids_by_z_level[center_turf.z]
+	switch(type)
+		if(SPATIAL_GRID_CONTENTS_TYPE_CLIENTS)
+			for(var/list/grid_row as anything in grid_level.Copy(min_y, max_y + 1))//from testing, slightly faster than iterating numbers from min_index to max_index
+				for(var/datum/spatial_grid_cell/cell as anything in grid_row.Copy(min_x, max_x + 1))
+					contents_to_return += cell.client_contents
```

Comment:
🤔

Reply:
im pretty sure it is since vars is also a var, so youre doing a GetVar then pushing the name then a ListGet instead of just a GetVar. at least according to crucible
![Screenshot_1377](https://user-images.githubusercontent.com/15794172/136124388-634bd39b-4946-459b-b986-963591f78bbe.png)
![Screenshot_1376](https://user-images.githubusercontent.com/15794172/136124392-c9dff06f-8756-4a87-b38c-6f4146c494ee.png)